Fix for bug when discount from sales rule "Fixed amount discount for whole cart" wasn't correctly applied for bundle products with dynamic price.

Commit: https://github.com/magento/magento2/commit/d56db5bed6424e94fb70ae116eb89360465d1762#diff-8ea41e8cf43e731ac4c8238cb2713466935f10578b659f3a0cb5ec96e5823919

diff --git a/Model/Rule/Action/Discount/CartFixed.php b/Model/Rule/Action/Discount/CartFixed.php
index 1569c95..8c1aadc 100644
--- a/Model/Rule/Action/Discount/CartFixed.php
+++ b/Model/Rule/Action/Discount/CartFixed.php
@@ -111,7 +111,9 @@ class CartFixed extends AbstractDiscount
                             $address,
                             $baseRuleTotals
                         ) : $baseRuleTotals;
-                $availableDiscountAmount = $this->cartFixedDiscountHelper
+                // BEGINNING OF PATCH
+                // Fix for bug when discount from sales rule "Fixed amount discount for whole cart" wasn't correctly applied for bundle products with dynamic price.
+                $maximumItemDiscount = $this->cartFixedDiscountHelper
                     ->getDiscountAmount(
                         $ruleDiscount,
                         $qty,
@@ -119,8 +121,9 @@ class CartFixed extends AbstractDiscount
                         $baseRuleTotals,
                         $discountType
                     );
-                $quoteAmount = $this->priceCurrency->convert($availableDiscountAmount, $store);
-                $baseDiscountAmount = min($baseItemPrice * $qty, $availableDiscountAmount);
+                $quoteAmount = $this->priceCurrency->convert($maximumItemDiscount, $store);
+                $baseDiscountAmount = min($baseItemPrice * $qty, $maximumItemDiscount);
+                // END OF PATCH
                 $this->deltaPriceRound->reset($discountType);
             } else {
                 $baseRuleTotals = $shippingMethod ?
